include/rmm/mr/device/detail/stream_ordered_memory_resource.hpp

```diff
@@ -244,10 +244,18 @@ class stream_ordered_memory_resource : public crtp<PoolResource>, public device_
     bytes        = rmm::detail::align_up(bytes, allocation_alignment);
     auto const b = this->underlying().free_block(p, bytes);
 
-    // TODO: cudaEventRecord has significant overhead on deallocations. For the non-PTDS case
-    // we may be able to delay recording the event in some situations. But using events rather than
-    // streams allows stealing from deleted streams.
-    RMM_ASSERT_CUDA_SUCCESS(cudaEventRecord(stream_event.event, stream));
+#ifdef CUDA_API_PER_THREAD_DEFAULT_STREAM
+    if (stream != cudaStreamLegacy) {
```

Comment:
I don't think this optimization is correct unless you change the call to `cudaStreamWaitEvent` in the same cases to a `cudaStreamSynchronize`.  I also think that this optimization is orthogonal to the present PR (it has been on my list to add, but not without benchmarking). Consider treating it in a separate PR.

Reply:
Removed.